### PR TITLE
Use react@next and react-dom@next

### PR DESCRIPTION
**Pull request by rogeliog**
I might be wrong here 😄 

The reason why there is a hardcoded `node_modules/react` and `node_modules/react-dom` is because of Fiber right? I changed it to use `react@next` which is now `16-alpha`.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -30,5 +30,9 @@
     "predeploy": "npm run build",
     "deploy": "gh-pages -d build",
     "prepublish": "npm run build"
+  },
+  "dependencies": {
```
Comment (rogeliog) [inline package.json]: Should this just be a dev dependency?

Comment (camwest) [reply package.json]: Yeah the pattern I've been using is to make react a peer and a dev dependency. Can you swap it to that?

**Review by camwest — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (rogeliog) [reply package.json]: I agree but right now the demo is inside of the app, so `npm start` depends on having access to `react`... Maybe extracting the demo directory might be a good solution, with this we can add it as `peerDependency` to `react-slot-fill` and as `dependency` to the demo.

Comment (camwest) [reply package.json]: Why not just add it as a peer/dev dependency and then the demo will still build locally webpack doesn't care if something is a dev or real dependency. 

Then when you npm install react-slot-fill it won't try to bring React with it...

I agree in the long run with moving the demos but in the short run I think it's no big deal.

Comment (rogeliog) [reply package.json]: Awesome!

**Review by camwest — APPROVED** [review]

**Pull request merged**